internal/iapl/policy.go

```diff
@@ -315,26 +647,106 @@ func (v *policy) Schema() []types.ResourceType {
 	}
 
 	for _, b := range v.bn {
+		actionName := b.ActionName
+
+		if b.RenamePermission != "" {
+			actionName = b.RenamePermission
+		}
+
 		action := types.Action{
-			Name: b.ActionName,
+			Name: actionName,
 		}
 
 		for _, c := range b.Conditions {
-			condition := types.Condition{
-				RoleBinding:        (*types.ConditionRoleBinding)(c.RoleBinding),
-				RelationshipAction: (*types.ConditionRelationshipAction)(c.RelationshipAction),
+			var conditions []types.Condition
+
+			switch {
+			case c.RoleBinding != nil:
```

Comment:
Can we break these conditions out so they're easier to read? This switch statement is really long.